psi4/src/psi4/libtrans/fcidump_helper.cc

```diff
@@ -0,0 +1,214 @@
+/*
+ * @BEGIN LICENSE
+ *
+ * Psi4: an open-source quantum chemistry software package
+ *
+ * Copyright (c) 2007-2017 The Psi4 Developers.
+ *
+ * The copyrights for code used from other parties are included in
+ * the corresponding files.
+ *
+ * This file is part of Psi4.
+ *
+ * Psi4 is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * Psi4 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with Psi4; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * @END LICENSE
+ */
+
+#include "fcidump_helper.h"
+
+#include <cmath>
+#include <cstdio>
+#include <map>
+#include <memory>
+#include <string>
+
+#include "psi4/psi4-dec.h"
+#include "psi4/psifiles.h"
+
+#include "psi4/libciomr/libciomr.h"
+#include "psi4/libdpd/dpd.h"
+#include "psi4/libmints/matrix.h"
+#include "psi4/libmints/wavefunction.h"
+#include "psi4/libpsi4util/PsiOutStream.h"
+#include "psi4/libpsio/psio.hpp"
+#include "psi4/libqt/qt.h"
+#include "psi4/libtrans/integraltransform.h"
+
+namespace psi {
+namespace fcidump {
+void fcidump_tei_helper(int nirrep, bool restricted, std::map<std::string, int> DPD_info, double ints_tolerance,
+                        std::string fname) {
+    outfile->Printf("Writing TEI integrals in FCIDUMP format to " + fname + "\n");
+    // Append to the file created by the fcidump function Python-side
+    FILE* intdump = fopen(fname.c_str(), "a");
```

Comment:
Should this code create a `shared_ptr<PsiOutStream>` instead of using a raw `FILE*`?

Reply:
Great suggestion. I've implemented as you suggest now.